AirCasting/Graph/GraphView.swift

```diff
@@ -36,35 +36,92 @@ struct GraphView<StatsViewModelType>: View where StatsViewModelType: StatisticsC
+                            if let selectedStream = selectedStream {
+                                Graph(stream: selectedStream,
+                                      thresholds: threshold,
+                                      isAutozoomEnabled: session.type == .mobile).onDateRangeChange { [weak graphStatsDataSource, weak statsContainerViewModel] range in
+                                        graphStatsDataSource?.dateRange = range
+                                        statsContainerViewModel?.adjustForNewData()
+                                      }
+                                StatisticsContainerView(statsContainerViewModel: statsContainerViewModel,
+                                                        threshold: threshold)
+                            }
 
-                }
-                NavigationLink(destination: HeatmapSettingsView(changedThresholdValues: threshold.rawThresholdsBinding)) {
-                    EditButtonView()
-                }
-                .padding(.horizontal)
-                .padding(.top)
-                
-                ThresholdsSliderView(threshold: threshold)
-                    .padding()
-                    // Fixes labels covered by tabbar
-                    .padding(.bottom)
+                        }
+                        HStack() {
+                            startTimeText
+                            Spacer()
+                            endTimeText
+                        }
+                        .foregroundColor(.aircastingGray)
+                        .padding(.horizontal, 5)
+                        NavigationLink(destination: HeatmapSettingsView(changedThresholdValues: threshold.rawThresholdsBinding)) {
+                            EditButtonView()
+                        }
+                        .padding(.horizontal)
+                        .padding(.top)
+                        
+                        ThresholdsSliderView(threshold: threshold)
+                            .padding()
+                            // Fixes labels covered by tabbar
+                            .padding(.bottom)
+                    }
             }
             Spacer()
         }
         .navigationBarTitleDisplayMode(.inline)
     }
+    
+    var startTimeText: some View {
+        startTime()
+    }
+    
+    var endTimeText: some View {
+        endTime()
+    }
+    
+    func procceding(session: SessionEntity) -> Bool {
+        var showGraph = false
+        guard session.allStreams != nil else { return false }
+        (session.allStreams!.forEach({ stream in
+            (stream.allMeasurements != []) ? (showGraph = true) : (showGraph = false)
+        }))
+        if showGraph {
+            return true
+        } else {
+            return false
+        }
+    }
+    
+    func startTime() -> Text {
+            let formatter = DateFormatter()
+            formatter.locale = Locale(identifier: "en_US")
+        if !(session.isMobile && session.isActive) {
```

Comment:
1. This feels magical, I know why it's here but if someone sees it for the first time they will have absolutely no clue why it's like that and will probably not be able to figure it out (so we're harming the flexibility to change the app, which is a core value of a software product). Please at least add a comment
2. Formatting seems off